src/api/map.js

```diff
@@ -36,7 +36,9 @@ export default class Map {
     }
 
     addLayer(layer, beforeLayerID) {
-        layer.requestData();
+        setTimeout(() => {
```

Comment:
hacky smell ? 🙄

Reply:
Right :joy:

Reply:
waiting for a good explanation ¬¬

Reply:
Now the `layer.setStyle` is an async function so in this case when `addLayer` calls to the `layer.initCallback` function the style doesn't exist at that time and the function crashes. I'm open to suggestions.